fragmenter/utils.py

```diff
@@ -1,179 +1,82 @@
 
 
-"""
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-Movie making functions
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-"""
+def get_fgroup_smarts() -> Dict[str, str]:
+    """Returns a dictionary containing the SMARTS representations of different
+    functional groups loaded from the internal ``fgroup_smarts.yml`` file.
 
-
-def merge_images(file1, file2, filename):
-    """Merge two images into one, displayed side by side
-    :param file1: path to first image file
-    :param file2: path to second image file
-    :return: the merged Image object
+    Returns
+    -------
+        A dictionary where each key is the name of a functional group and each value
+        the corresponding SMARTS pattern.
     """
-    try:
-        from PIL import Image
-    except ImportError:
-        raise ImportError("Must have PIL installed to use this function")
 
-    image1 = Image.open(file1)
-    image2 = Image.open(file2)
+    file_name = resource_filename(
+        "fragmenter", os.path.join("data", "fgroup_smarts.yml")
+    )
 
-    (width1, height1) = image1.size
-    (width2, height2) = image2.size
+    with open(file_name, "r") as file:
+        functional_groups = yaml.safe_load(file)
 
-    result_width = width1 + width2 - 500 # play with the number here
-    result_height = max(height1, height2)
+    return functional_groups
 
-    result = Image.new('RGB', (result_width, result_height), color='white')
-    result.paste(im=image2, box=(width1-250, 0)) # play with the number here
-    result.paste(im=image1, box=(0, int(result_height/2) - int(height1/2)))
-    result.save(filename)
 
+def get_map_index(
+    molecule: Molecule, atom_index: int, error_on_missing: bool = True
+) -> int:
+    """Returns the map index of a particular atom in a molecule.
 
-def make_movie(files, fps=15, filename='movie'):
-    """
-    Make movie from list of files
-
-    Parameters
-    ----------
-    files: list of paths to files
-    fps: files per second, int, optional
-        defualt 15
-    filename: str, optional
-        default 'movie
-
-    """
-    try:
-        import moviepy.editor as mpy
-    except ImportError:
-        raise ImportError("You need to install moviepy to use this function")
-
-    im = mpy.ImageSequenceClip(files, fps=fps)
-    im.write_videofile('{}.mp4'.format(filename), fps=fps)
-
-def serialize_bond(key):
-    """
-    Serialize bond tuple for JSON
     Parameters
     ----------
-    key : tuple of ints
+    molecule
+        The molecule containing the atom.
+    atom_index
+        The index of the atom in the molecule.
+    error_on_missing
+        Whether an error should be raised if the atom does not have a corresponding
+        map index
 
     Returns
     -------
-    serialized tuple
-
+        The map index if found, otherwise 0.
     """
-    if isinstance(key, (int, float)):
-        key = (int(key), )
+    atom_map = molecule.properties.get("atom_map", {})
+    atom_map_index = atom_map.get(atom_index, None)
```

Comment:
> it looks like you set the atom_map property yourself and all are ints

The atom map comes from the OpenFF toolkit itself and so in general we cannot make too many assumptions here. 

> I bring this up just because get_map_index is used a lot, in a number of nested loops.

I agree this implementation is quite lazy but its cost should be effectively hidden by the cost of expensive conformer generation / AM1 calculations etc, and so I don't think we need to worry about the performance just yet until we've done some profiling to see how much of a bottleneck this is!